server/serverimpl.go

```diff
@@ -292,22 +296,22 @@ func (s *server) handlePlainHTTPRequest(req *http.Request, w http.ResponseWriter
 		conn: connFromRequest(req),
 	}
 
-	if s.settings.Callbacks == nil {
+	if connectionHandler == nil {
```

Comment:
We probably need to document somewhere that the connection callback is required for plain http connections.